js/hw_3_js_react_1/src/utils.js

```diff
@@ -0,0 +1,100 @@
+export const getClassName = (rowId,cellId,snake,food,extraFood) => {
+    let className = 'cell';
+    for(let i = 0; i < snake.length;i++){
+      if(snake[i][0] === rowId && snake[i][1] === cellId){
+        className = 'cell snake';
+      }
+    }
+    
+    if(food[0] === rowId && food[1] === cellId){
+      className = 'cell food';
+    }
+     
+    
+    if(extraFood !== undefined && extraFood[0] === rowId && extraFood[1] === cellId){
+      className  = 'cell bonus-food';
+    }
+  
+  
+    return className;
+  };
+
+
+  export function generateFood(snakeBody,food) {
+    for(let i = 0;i < snakeBody.length; i++){
+        if(snakeBody[i][0] !== food[0] && snakeBody[i][0] !== food[1]){
+            food = [Math.floor(Math.random() * 24),Math.floor(Math.random() * 24)];
+           
+           
+        }
+    }
+    return food;
+}
+
+export function generateExtraFood(snakeBody,food,extraFood){
+    for(let i = 0;i < snakeBody.length; i++){
+        if(extraFood === undefined || (snakeBody[i][0] !== extraFood[0] && snakeBody[i][0] !== extraFood[1] && extraFood[0] !== food[0] && extraFood[1] !== food[1])){
```

Comment:
This code is **very hard** to maintain. Do not be shy - create additional constants, and make it human-readable.